### PR TITLE
PCHR-3467: Workflow form changes

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview

This PR adds the following fixes to the workflow form:

* It aligns the activity fields to the top of their table container
* It adds a scrollbar to the tabs panel in case the panel's content overflows
* Fixes a wrong selector introduced in https://github.com/compucorp/org.civicrm.shoreditch/pull/92

## Before
anim

## After
anim

### Overflow scroll and elements width

To allow the overflow content to be scrolled, an `scroll-x: auto` property was added to the ui tabs panels element:

```scss
.ui-tabs-panel {
  display: table;
  overflow-x: auto;
  width: 100%;
  // ...

  > * {
    display: table-row;
  }
}
```

The `display: table` and `display: table-row` was added because elements inside overflown containers don't take the full width of the content inside the container.

For example, if the container is 1,000px wide and there's an element A with a width of 1,500px and another element B with  a width of 500px, if you set the width of element B to be 100% it would be 1,000px instead of 1,500px. This is an issue because we'd like the "advanced" accordion to be as wide as the table for activities.

By setting these two elements to behave as table rows they take the width of the widest element.

### Fixing the selector for the activity table

Previously the selector used for the activity table was `table:first-child` this was not right since the table is not the first child of the `.ui-tabs-panel` container. To properly select the activity table one must use `* table:first-child`

### Aligning the elements inside the activity table

```scss
* table:first-child {
  // ...

  tr > td {
    vertical-align: top;
  }
}
```

### CODE REVIEW

File: scss/civicrm/administer/civi-case/_case-types.scss [hunk 1]
```diff
@@ -34,10 +34,23 @@
...
       .ui-tabs-panel {
+        display: table;
+        overflow-x: auto;
         padding: 0 !important;
+        width: 100%;
 
-        table:first-child {
+        // Makes sure any element inside the panel has a true width 100% depending
+        // on the widest sibling and not the panel's width:
+        > * {
```
Comment (AkA84) [inline scss/civicrm/administer/civi-case/_case-types.scss]: `*` should be used only if really necessary, is it the case here? You don't know what elements in particular you want to behave like a table row?

File: scss/civicrm/administer/civi-case/_case-types.scss [hunk 2]
```diff
@@ -34,10 +34,23 @@
...
+          display: table-row;
+        }
+
+        * table:first-child {
```
Comment (AkA84) [inline scss/civicrm/administer/civi-case/_case-types.scss]: same as above for `*`, why is it needed? Doesn't the table have always the same parent element?

File: scss/civicrm/administer/civi-case/_case-types.scss [hunk 3]
```diff
@@ -34,10 +34,23 @@
...
           border-top: 0;
+
+          tr > td {
```
Comment (AkA84) [inline scss/civicrm/administer/civi-case/_case-types.scss]:
```scss
tr {
 > td {
 }
}
```
or can't you simply do
```scss
td {

}
```
? Why is the `tr >` necessary?

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: scss/civicrm/administer/civi-case/_case-types.scss (hunk 3 above)
Comment (reneolivo) [reply scss/civicrm/administer/civi-case/_case-types.scss]: `tr >` was not necessary, it was just me being silly.

File: scss/civicrm/administer/civi-case/_case-types.scss (hunk 1 above)
Comment (reneolivo) [reply scss/civicrm/administer/civi-case/_case-types.scss]: I know the specific elements, but one of them is a div with no discerning class or attribute to use as a selector. I changed it to this:

```scss
> div,
> .crm-accordion-wrapper {
  // ...
}
```

**Comment by reneolivo** [comment]
@AkA84 [this commit](https://github.com/compucorp/org.civicrm.shoreditch/pull/98/commits/42ce7a4ace824cf382a249188de81b1f102640d1) was added because the children would not have a 100% width when their parent had no scrolls depending on the screen's width.

To fix it the display of the element is reverted back to their normal state when the screen's width is bigger than 1200px (`$screen-lg`), which makes the child elements display with a width of 100%

**Review by AkA84 — APPROVED** [review]

**Pull request merged**